### ISSUE TITLE
Problem with ./choose

### ISSUE THREAD

**Issue by kasadi97** [ZeGentzy/hookey#33]
Hello! Right now I finally understood how to get SSH key and completed Part 1(git submodule update --init --recursive) and started doing part 2. Here I am stacked at part 2. After I run ./choose 1, there is a lot of lines generated in terminal (i think it should not happen).

`+/etc/zsh/zshenv:15> [[ -z /home/adilbek/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin || /home/adilbek/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin == /bin:/usr/bin ]]
+./choose:3> echo 1
+./choose:5> source ./common
+./common:3> var_keys=( INSTALLS INSTALL INSTALL_STR APP APP_ROOT APP_MARK_EXEC_LOC APP_MARK_EXEC ) 
+./common:13> declare -A var_types=([$'\M-"'INSTALLS[file$'\M-"'[INSTALLfile[$'\M-"'INSTALL_STR[eval$'\M-"'[APPeval[$'\M-"'APP_ROOT[eval$'\M-"'[APP_MARK_EXEC_LOCeval[$'\M-"'APP_MARK_EXEC )
+./common:24> declare -A vars=([$'\M-"'INSTALLS['"./.installs"'$'\M-"'[INSTALL'"./.install"'[$'\M-"'INSTALL_STR['"$(echo "$INSTALLS" | head -n"$(( $INSTALL + 0 ))" | tail -n1)"'$'\M-"'[APP'"$(echo "$INSTALL_STR" | cut -d" " -f1)"'[$'\M-"'APP_ROOT['"$(echo "$INSTALL_STR" | cut -d" " -f2-)"'$'\M-"'[APP_MARK_EXEC_LOC'"$(pwd)/apps/${APP}/executables"'[$'\M-"'APP_MARK_EXEC )
+./common:34> declare -A var_deps=([$'\M-"'INSTALLS['()'$'\M-"'[INSTALL'()'[$'\M-"'INSTALL_STR['(INSTALLS INSTALL)'$'\M-"'[APP'(INSTALL_STR)'[$'\M-"'APP_ROOT['(INSTALL_STR)'$'\M-"'[APP_MARK_EXEC_LOC'(APP)'[$'\M-"'APP_MARK_EXEC )
+./common:45> declare -A vars_done=( )
+./common:72> populate
]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=]=+populate:1> key=INSTALLS
+populate:2> eval 'local this_deps=()'
+(eval):1> local this_deps=( )
+populate:8> local var_type=file
+populate:9> local var='"./.installs"'
+populate:10> [[ file == file ]]
+populate:11> eval 'local file="./.installs"'
+(eval):1> local file=./.installs
+populate:12> [[ -f ./.installs ]]
+populate:13> cat ./.installs
+populate:13> export $'INSTALLS=ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II\nck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+populate:14> vars_done[$key]=1 
+populate:1> key=INSTALL
+populate:2> eval 'local this_deps=()'
+(eval):1> local this_deps=( )
+populate:8> local var_type=file
+populate:9> local var='"./.install"'
+populate:10> [[ file == file ]]
+populate:11> eval 'local file="./.install"'
+(eval):1> local file=./.install
+populate:12> [[ -f ./.install ]]
+populate:13> cat ./.install
+populate:13> export 'INSTALL=1'
+populate:14> vars_done[$key]=1 
+populate:1> key=INSTALL_STR
+populate:2> eval 'local this_deps=(INSTALLS INSTALL)'
+(eval):1> local this_deps=( INSTALLS INSTALL )
+populate:3> dep=INSTALLS
+populate:4> [[ 1 != 1 ]]
+populate:3> dep=INSTALL
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='"$(echo "$INSTALLS" | head -n"$(( $INSTALL + 0 ))" | tail -n1)"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export INSTALL_STR="$(echo "$INSTALLS" | head -n"$(( $INSTALL + 0 ))" | tail -n1)"'
+(eval):1> echo $'ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II\nck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+(eval):1> head -n1
+(eval):1> tail -n1
+(eval):1> export INSTALL_STR='ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+populate:20> vars_done[$key]=1 
+populate:1> key=APP
+populate:2> eval 'local this_deps=(INSTALL_STR)'
+(eval):1> local this_deps=( INSTALL_STR )
+populate:3> dep=INSTALL_STR
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='"$(echo "$INSTALL_STR" | cut -d" " -f1)"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export APP="$(echo "$INSTALL_STR" | cut -d" " -f1)"'
+(eval):1> echo 'ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+(eval):1> cut '-d ' -f1
+(eval):1> export APP=ck2
+populate:20> vars_done[$key]=1 
+populate:1> key=APP_ROOT
+populate:2> eval 'local this_deps=(INSTALL_STR)'
+(eval):1> local this_deps=( INSTALL_STR )
+populate:3> dep=INSTALL_STR
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='"$(echo "$INSTALL_STR" | cut -d" " -f2-)"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export APP_ROOT="$(echo "$INSTALL_STR" | cut -d" " -f2-)"'
+(eval):1> echo 'ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+(eval):1> cut '-d ' -f2-
+(eval):1> export APP_ROOT='/Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+populate:20> vars_done[$key]=1 
+populate:1> key=APP_MARK_EXEC_LOC
+populate:2> eval 'local this_deps=(APP)'
+(eval):1> local this_deps=( APP )
+populate:3> dep=APP
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='"$(pwd)/apps/${APP}/executables"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export APP_MARK_EXEC_LOC="$(pwd)/apps/${APP}/executables"'
+(eval):1> pwd
+(eval):1> export APP_MARK_EXEC_LOC=/home/adilbek/hookey/apps/ck2/executables
+populate:20> vars_done[$key]=1 
+populate:1> key=APP_MARK_EXEC
+populate:2> eval 'local this_deps=(APP_MARK_EXEC_LOC)'
+(eval):1> local this_deps=( APP_MARK_EXEC_LOC )
+populate:3> dep=APP_MARK_EXEC_LOC
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=file
+populate:9> local var='"$APP_MARK_EXEC_LOC"'
+populate:10> [[ file == file ]]
+populate:11> eval 'local file="$APP_MARK_EXEC_LOC"'
+(eval):1> local file=/home/adilbek/hookey/apps/ck2/executables
+populate:12> [[ -f /home/adilbek/hookey/apps/ck2/executables ]]
+populate:13> cat /home/adilbek/hookey/apps/ck2/executables
+populate:13> export $'APP_MARK_EXEC=./dbghelp.dll\n./libtbbmalloc.so.2\n./CK2game.exe\n./steamclient.so\n./tbb.dll\n./tbbmalloc_proxy.dll\n./pthreadVCE2.dll\n./libpops_api.so\n./Steam.dll\n./pops_api.dll\n./steamclient.dll\n./steam_api.dll\n./PDXBrowser_IPC.dll\n./pdx_browser/CEFProcess.exe\n./pdx_browser/widevinecdmadapter.dll\n./pdx_browser/PDXBrowser.dll\n./pdx_browser/CEFProcess\n./pdx_browser/libGLESv2.dll\n./pdx_browser/libcef.so\n./pdx_browser/libcef.dll\n./pdx_browser/d3dcompiler_47.dll\n./pdx_browser/chrome-sandbox\n./pdx_browser/libEGL.dll\n./pdx_browser/PDXBrowser.so\n./pdx_browser/libwidevinecdmadapter.so\n./pdx_browser/d3dcompiler_43.dll\n./tbbmalloc.dll\n./pdx_online.dll\n./libsteam.so\n./libpdx_online.so\n./ck2\n./PDXBrowser_IPC.so\n./libtbb.so.2\n./PdxConnect.dll\n./libtbbmalloc_proxy.so.2\n./tbb_debug.dll\n./lua51.dll\n./libsteam_api.so\n./lua5.1.dll'
+populate:14> vars_done[$key]=1 
+./common:74> [[ ck2 !=  ]]
+./common:75> [[ ! -d ./apps/ck2 ]]
+./common:80> source ./apps/ck2/common
+./apps/ck2/common:3> var_keys+=( OVERRIDE_DIRS GOLDBERG_BUILD ) 
+./apps/ck2/common:5> vars[OVERRIDE_DIRS]='("${APP_ROOT}")' 
+./apps/ck2/common:6> var_types[OVERRIDE_DIRS]=eval 
+./apps/ck2/common:7> var_deps[OVERRIDE_DIRS]='(APP_ROOT)' 
+./apps/ck2/common:9> vars[GOLDBERG_BUILD]='"release32"' 
+./apps/ck2/common:10> var_types[GOLDBERG_BUILD]=eval 
+./apps/ck2/common:11> var_deps[GOLDBERG_BUILD]='()' 
+./common:81> populate
+populate:1> key=INSTALLS
+populate:2> eval 'local this_deps=()'
+(eval):1> local this_deps=( )
+populate:8> local var_type=file
+populate:9> local var='"./.installs"'
+populate:10> [[ file == file ]]
+populate:11> eval 'local file="./.installs"'
+(eval):1> local file=./.installs
+populate:12> [[ -f ./.installs ]]
+populate:13> cat ./.installs
+populate:13> export $'INSTALLS=ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II\nck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+populate:14> vars_done[$key]=1 
+populate:1> key=INSTALL
+populate:2> eval 'local this_deps=()'
+(eval):1> local this_deps=( )
+populate:8> local var_type=file
+populate:9> local var='"./.install"'
+populate:10> [[ file == file ]]
+populate:11> eval 'local file="./.install"'
+(eval):1> local file=./.install
+populate:12> [[ -f ./.install ]]
+populate:13> cat ./.install
+populate:13> export 'INSTALL=1'
+populate:14> vars_done[$key]=1 
+populate:1> key=INSTALL_STR
+populate:2> eval 'local this_deps=(INSTALLS INSTALL)'
+(eval):1> local this_deps=( INSTALLS INSTALL )
+populate:3> dep=INSTALLS
+populate:4> [[ 1 != 1 ]]
+populate:3> dep=INSTALL
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='"$(echo "$INSTALLS" | head -n"$(( $INSTALL + 0 ))" | tail -n1)"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export INSTALL_STR="$(echo "$INSTALLS" | head -n"$(( $INSTALL + 0 ))" | tail -n1)"'
+(eval):1> echo $'ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II\nck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+(eval):1> head -n1
+(eval):1> tail -n1
+(eval):1> export INSTALL_STR='ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+populate:20> vars_done[$key]=1 
+populate:1> key=APP
+populate:2> eval 'local this_deps=(INSTALL_STR)'
+(eval):1> local this_deps=( INSTALL_STR )
+populate:3> dep=INSTALL_STR
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='"$(echo "$INSTALL_STR" | cut -d" " -f1)"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export APP="$(echo "$INSTALL_STR" | cut -d" " -f1)"'
+(eval):1> echo 'ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+(eval):1> cut '-d ' -f1
+(eval):1> export APP=ck2
+populate:20> vars_done[$key]=1 
+populate:1> key=APP_ROOT
+populate:2> eval 'local this_deps=(INSTALL_STR)'
+(eval):1> local this_deps=( INSTALL_STR )
+populate:3> dep=INSTALL_STR
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='"$(echo "$INSTALL_STR" | cut -d" " -f2-)"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export APP_ROOT="$(echo "$INSTALL_STR" | cut -d" " -f2-)"'
+(eval):1> echo 'ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+(eval):1> cut '-d ' -f2-
+(eval):1> export APP_ROOT='/Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
+populate:20> vars_done[$key]=1 
+populate:1> key=APP_MARK_EXEC_LOC
+populate:2> eval 'local this_deps=(APP)'
+(eval):1> local this_deps=( APP )
+populate:3> dep=APP
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='"$(pwd)/apps/${APP}/executables"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export APP_MARK_EXEC_LOC="$(pwd)/apps/${APP}/executables"'
+(eval):1> pwd
+(eval):1> export APP_MARK_EXEC_LOC=/home/adilbek/hookey/apps/ck2/executables
+populate:20> vars_done[$key]=1 
+populate:1> key=APP_MARK_EXEC
+populate:2> eval 'local this_deps=(APP_MARK_EXEC_LOC)'
+(eval):1> local this_deps=( APP_MARK_EXEC_LOC )
+populate:3> dep=APP_MARK_EXEC_LOC
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=file
+populate:9> local var='"$APP_MARK_EXEC_LOC"'
+populate:10> [[ file == file ]]
+populate:11> eval 'local file="$APP_MARK_EXEC_LOC"'
+(eval):1> local file=/home/adilbek/hookey/apps/ck2/executables
+populate:12> [[ -f /home/adilbek/hookey/apps/ck2/executables ]]
+populate:13> cat /home/adilbek/hookey/apps/ck2/executables
+populate:13> export $'APP_MARK_EXEC=./dbghelp.dll\n./libtbbmalloc.so.2\n./CK2game.exe\n./steamclient.so\n./tbb.dll\n./tbbmalloc_proxy.dll\n./pthreadVCE2.dll\n./libpops_api.so\n./Steam.dll\n./pops_api.dll\n./steamclient.dll\n./steam_api.dll\n./PDXBrowser_IPC.dll\n./pdx_browser/CEFProcess.exe\n./pdx_browser/widevinecdmadapter.dll\n./pdx_browser/PDXBrowser.dll\n./pdx_browser/CEFProcess\n./pdx_browser/libGLESv2.dll\n./pdx_browser/libcef.so\n./pdx_browser/libcef.dll\n./pdx_browser/d3dcompiler_47.dll\n./pdx_browser/chrome-sandbox\n./pdx_browser/libEGL.dll\n./pdx_browser/PDXBrowser.so\n./pdx_browser/libwidevinecdmadapter.so\n./pdx_browser/d3dcompiler_43.dll\n./tbbmalloc.dll\n./pdx_online.dll\n./libsteam.so\n./libpdx_online.so\n./ck2\n./PDXBrowser_IPC.so\n./libtbb.so.2\n./PdxConnect.dll\n./libtbbmalloc_proxy.so.2\n./tbb_debug.dll\n./lua51.dll\n./libsteam_api.so\n./lua5.1.dll'
+populate:14> vars_done[$key]=1 
+populate:1> key=OVERRIDE_DIRS
+populate:2> eval 'local this_deps=(APP_ROOT)'
+(eval):1> local this_deps=( APP_ROOT )
+populate:3> dep=APP_ROOT
+populate:4> [[ 1 != 1 ]]
+populate:8> local var_type=eval
+populate:9> local var='("${APP_ROOT}")'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export OVERRIDE_DIRS=("${APP_ROOT}")'
+(eval):1> export OVERRIDE_DIRS=( '/Data/Games/SteamLibrary/steamapps/common/Crusader Kings II' )
+populate:20> vars_done[$key]=1 
+populate:1> key=GOLDBERG_BUILD
+populate:2> eval 'local this_deps=()'
+(eval):1> local this_deps=( )
+populate:8> local var_type=eval
+populate:9> local var='"release32"'
+populate:10> [[ eval == file ]]
+populate:18> [[ eval == eval ]]
+populate:19> eval 'export GOLDBERG_BUILD="release32"'
+(eval):1> export GOLDBERG_BUILD=release32
+populate:20> vars_done[$key]=1 
+./common:84> echo 'Populated INSTALL_STR INSTALL APP_MARK_EXEC INSTALLS APP_MARK_EXEC_LOC OVERRIDE_DIRS APP_ROOT GOLDBERG_BUILD APP'
Populated INSTALL_STR INSTALL APP_MARK_EXEC INSTALLS APP_MARK_EXEC_LOC OVERRIDE_DIRS APP_ROOT GOLDBERG_BUILD APP
+./choose:7> echo 'ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II'
ck2 /Data/Games/SteamLibrary/steamapps/common/Crusader Kings II `

What did i do wrong, possibly? Or what do i miss? TY in advance!

**Comment by kasadi97** [comment]
I tried to install clang, but result is the same

**Comment by kasadi97** [comment]
Holy cow! It took me some time and tries to figure out how to do and what to do from the scratch. So, for all noobs like me:
Wall of text in Terminal is okay. Look at the end of the text! If there is errors try to google the solution!
In my situation I preinstalled clang, as i saw it might be problem. Also you will need sudo apt-get install gcc-multilib and sudo apt-get install g++-multilib. Also Goldberg require its OWN dependencies. Which are: protobuf-lite (the dev package) and protoc. It means u will need to -> sudo apt install libprotobuf-dev and sudo apt install protobuf-compiler. 

In my case I installed some wrong things too, but in the end I figure out the most important things and their are listed above.

At the end, you will get error in the ./run, but do not get too depressed. Try to run game launcher in steam, most probably all dlcs unlocked there!!!
